### DISCUSSION TITLE
Feature - Copy files to local storage

### DISCUSSION THREAD

**Discussion by gavim** [MoleMan1024/audiowagon#143] [Ideas]
Dear Contributors,
Thank you for making this amazing app!

Do you think it's possible to add a feature to support copying music files from flash drive to local storage?
This would allow us to listen to music without keeping the flash drive plugged in every time, so that we can use the port for charging or connecting to phones or other devices.

**Comment by MoleMan1024** [comment]
@gavim : This is technically possible, however I am bit concerned about how people might use that and that it could [wear out the internal flash memory](https://source.android.com/docs/core/perf/flash-wear). After all, replacing a headunit is more expensive than replacing a phone and cars are used for an average of about 12 years while the average phone is usually replaced after 2 to 4 years already. If car drivers copy stuff to internal disk all the time, this might become an issue.
Wearing out an external USB flash drive or SSD instead is safer from that point of view.

You should have two USB ports in the front though? Are there cars that have only one? Are you charging multiple phones at the same time?

**Comment by micksmothers** [comment]
This is an interesting request - for someone who has a relatively static playlist and with sufficient warnings on wear, perhaps this would be a nice feature. I agree that I don't need the free port for charging on my Polestar as there is another one plus two more in the back, however there is only one data port in the entire car. I haven't run into a situation where I've needed to hook up something additionally for data but if I did then I'd have to exit AudioWagon to use the data port. Maybe a low priority item in the backlog? My thoughts anyway.

**Comment by gavim** [comment]
Thanks sharing your thoughts! Yes, if we take into consideration the hardware wear, then it does make sense to prioritize flash drive.
Though I do share the same situation with Mick - my car has multiple charging ports, but only one of them is USB data port. Sometimes, I find myself unplug the flash drive just to connect to Carplay on my phone (unfortunately, my car also doesn't support wireless Carplay).
This is probably a very niche situation, so if you do consider this a part of the roadmap, maybe a very low priority one?
